### PR TITLE
[1.6] Add optimizer_for_mobile doc into python api root doc

### PR DESCRIPTION
**Pull request by jlin27**
PR in unstable (master): https://github.com/pytorch/pytorch/commit/04320a47d766e6535aa7442927e79f0d46257c9c

### CODE REVIEW

**Review by xcheng16 — APPROVED** [review]

File: docs/source/mobile_optimizer.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    - **ReLU/Hardtanh fusion**: XNNPACK ops support fusion of clamping. That is clamping of output activation is done as part of the kernel, including for 2D convolution and linear op kernels. Thus clamping effectively comes for free. Thus any op that can be expressed as clamping op, such as ``ReLU`` or ``hardtanh``, can be fused with previous ``Conv2D`` or ``linear`` op in XNNPACK. This pass rewrites graph by finding ``ReLU/hardtanh`` ops that follow XNNPACK ``Conv2D/linear`` ops, written by the previous pass, and fuses them together.
+    - **Dropout removal** (blacklisting option `MobileOptimizerType::REMOVE_DROPOUT`): This optimization pass removes ``dropout`` and ``dropout_`` nodes from this module when training is false.
+
+``optimize_for_mobile`` will also invoke freeze_module pass which only preserves ``forward`` method. If you have other method to that needed to be preserved,  add them into the preserved method list and pass into the method.
```
Comment (malfet) [inline docs/source/mobile_optimizer.rst:16]:
```suggestion
``optimize_for_mobile`` will also invoke freeze_module pass which only preserves ``forward`` method. 
```

File: torch/utils/mobile_optimizer.py [hunk 2]
```diff
@@ -5,23 +5,27 @@
 import torch
 from enum import Enum
 from torch._C import MobileOptimizerType
-from typing import Set
+from typing import Set, List, AnyStr
```
Comment (malfet) [inline torch/utils/mobile_optimizer.py:8]:
```suggestion
from typing import Set
```

File: torch/utils/mobile_optimizer.py [hunk 3]
```diff
@@ -5,23 +5,27 @@
...
     BUNDLED_INPUT = 1
     REQUIRES_GRAD = 2
     DROPOUT = 3
     BATCHNORM = 4
 
-def optimize_for_mobile(script_module, optimization_blacklist: Set[MobileOptimizerType] = None):
+def optimize_for_mobile(
+        script_module,
+        optimization_blacklist: Set[MobileOptimizerType] = None,
+        preserved_methods: List[AnyStr] = None):
```
Comment (malfet) [inline torch/utils/mobile_optimizer.py:19]:
```suggestion
def optimize_for_mobile(script_module, optimization_blacklist: Set[MobileOptimizerType] = None):
```

File: torch/utils/mobile_optimizer.py [hunk 4]
```diff
@@ -5,23 +5,27 @@
...
     Args:
-        script_module: An instance of torch script module with type of ScriptModule
-        optimization_blacklist: A set with type of MobileOptimizerType.
-        When set is not passed, optimization method will run all the optimizer pass; otherwise, optimizer
-        method will run the optimization pass that is not included inside optimization_blacklist.
+        script_module: An instance of torch script module with type of ScriptModule.
+        optimization_blacklist: A set with type of MobileOptimizerType. When set is not passed,
+            optimization method will run all the optimizer pass; otherwise, optimizer
+            method will run the optimization pass that is not included inside optimization_blacklist.
+        perserved_methods: A list of methods that needed to be preserved when freeze_module pass is invoked.
```
Comment (malfet) [inline torch/utils/mobile_optimizer.py:26]:
```suggestion
```

File: torch/utils/mobile_optimizer.py [hunk 5]
```diff
@@ -30,7 +34,10 @@ def optimize_for_mobile(script_module, optimization_blacklist: Set[MobileOptimiz
     if optimization_blacklist is None:
         optimization_blacklist = set()
 
-    optimized_cpp_module = torch._C._jit_pass_optimize_for_mobile(script_module._c, optimization_blacklist)
+    if preserved_methods is None:
+        preserved_methods = []
+
+    optimized_cpp_module = torch._C._jit_pass_optimize_for_mobile(script_module._c, optimization_blacklist, preserved_methods)
```
Comment (malfet) [inline torch/utils/mobile_optimizer.py:40]:
```suggestion
    optimized_cpp_module = torch._C._jit_pass_optimize_for_mobile(script_module._c, optimization_blacklist)
```

**Review by malfet — APPROVED** [review]
LGTM is CI is green

**Pull request merged**